packages/component-library/src/CardList/CardList.js

```diff
@@ -158,25 +81,28 @@ const CardList = ({ CardRegistry, tagsList, projects }) => {
   const classes = useStyles();
   const theme = useTheme();
 
+  // Nested List and Drawer open / close handlers
   const [showAllStories, setShowAllStories] = useState(true);
   const [mobileOpen, setMobileOpen] = useState(false);
   const [openTopic, setOpenTopic] = useState(true);
   const [openLocation, setOpenLocation] = useState(false);
   const [openVisualization, setOpenVisualization] = useState(false);
+
+  // holds state for checkboxes (nested object where specific tags are boolean)
   const [filterStatus, setFilterStatus] = useState(allTagsFalse);
-  const [activeTagsList, setActiveTagsList] = useState([]);
-  const [activeTagsByCategory, setActiveTagsByCategory] = useState(
-    categoryNames.reduce(
-      (accumulator, categoryName) => ({
-        [categoryName]: [],
-        ...accumulator
-      }),
-      {}
-    )
-  );
+  // flat array of active tags
+  const activeTagsList = deriveActiveTagsListsFromFilterStatus(
+    filterStatus,
+    categoryNames
+  )[0];
+  // arrays of active tags separated by category
+  const activeTagsByCategory = deriveActiveTagsListsFromFilterStatus(
+    filterStatus,
+    categoryNames
+  )[1];
```

Comment:
We should avoid calling a function with the same arguments, and therefore returning the same outputs, multiple times. This becomes especially important when dealing with large amounts of data because of performance implications. It still matters though because there will be a higher maintenance cost if the `deriveActiveTagsListsFromFilterStatus` is changed in some way. It's pretty small now, but it is an unnecessary complexity. I would refactor this code to look something like:

```javascript
const deriveActiveTagsListsFromFilterStatusResults = deriveActiveTagsListsFromFilterStatus(
  filterStatus,
  categoryNames
);
// flat array of active tags
const activeTagsList = deriveActiveTagsListsFromFilterStatusResults[0];
// arrays of active tags separated by category
const activeTagsByCategory = deriveActiveTagsListsFromFilterStatusResults[1];
```